rest/multipart.go

```diff
@@ -27,30 +27,23 @@ const kMaxInlineAttachmentSize = 200
 // JSON bodies smaller than this won't be GZip-encoded.
 const kMinCompressedJSONSize = 300
 
-// Parses a JSON MIME body, unmarshaling it into "into".
+// ReadConfigJSON parses a JSON body, unmarshals it into the target structure "into".
 // Closes the input io.ReadCloser once done.
-func ReadJSONFromMIME(headers http.Header, input io.ReadCloser, into interface{}) error {
-	contentType := headers.Get("Content-Type")
-	if contentType != "" && !strings.HasPrefix(contentType, "application/json") {
-		return base.HTTPErrorf(http.StatusUnsupportedMediaType, "Invalid content type %s", contentType)
+func ReadConfigJSON(headers http.Header, input io.ReadCloser, into interface{}) error {
```

Comment:
This doesn't seem to have anything to do with a config, it looks like `readJSONInto`/`ReadJSONFromMIME` is used by a lot of different REST endpoints, not just for config.
The new name is actually less clear what it's doing. What was the rationale for the rename?

Reply:
This function seemed poorly named earlier, it was not reading any JSON from MIME other than just doing the content type and encoding check with the MIME header to "fail fast" and return the error message without spending unnecessary resources on parsing the body. But I agree with your point that the new name "ReadConfigJSON" is also not that great when considering the usage of this function in all other places aside from both handlePutDbConfig and handleCreateDB endpoints. Let’s keep the old name.